server/github_test.go

```diff
@@ -98,3 +115,223 @@ func TestCannotGetAllOrgMembersDueToRateLimit(t *testing.T) {
 
 	assert.Equal(t, originalUserSize, len(s.OrgMembers))
 }
+
+func TestCacheTransport(t *testing.T) {
+	t.Run("Should return cached response", func(t *testing.T) {
+		httpmock.Activate()
+		defer httpmock.DeactivateAndReset()
+
+		httpmock.RegisterResponder("GET", "https://api.github.com/repos/ownerTest/repoTest/git/ref/refTest",
+			func(req *http.Request) (*http.Response, error) {
+				body := &github.Reference{Object: &github.GitObject{}}
+				err := json.Unmarshal([]byte(testGetRefReturn), &body)
+				require.NoError(t, err)
+				resp, err := httpmock.NewJsonResponse(200, body)
```

Comment:
In this case, `resp` is not possible to be nil because we're providing the elements, so no need to check for nil here.

Changing literal 200 for `http.StatusOK` is a great suggestion and I've included it. Thank you!